dialer.go

```diff
@@ -9,19 +9,35 @@ import (
 // to the message bus.
 type Dialer func(string, string) (net.Conn, error)
 
-// DefaultDialer is the RPC server default implementation of
-// a dialer.
-func DefaultDialer(network, addr string) (net.Conn, error) {
-	conn, err := net.DialTimeout(network, addr, 2*time.Second)
-	if err != nil {
-		return nil, err
-	}
-	// Heartbeating hasn't started yet, don't stall forever on a dead server.
-	// A deadline is set for TLS and AMQP handshaking. After AMQP is established,
-	// the deadline is cleared in openComplete.
-	if err := conn.SetDeadline(time.Now().Add(2 * time.Second)); err != nil {
-		return nil, err
+// DialConf is a structure that you can pass to the default dialer to
+// change the defautl timeouts for dailing and handshakes.
+type DialConf struct {
+	DialTimeout time.Duration
+	Deadline    time.Duration
+}
+
+// DefaultDialer will return the default RPC server default implementation of
+// a dialer. With the DialConf parameter the default timeous can be specified.
+func DefaultDialer(config *DialConf) Dialer {
+	if config == nil {
+		config = &DialConf{
```

Comment:
Usually having a "zero" value of something means a reasonable default. Having a "zero" value for `Deadline` here would make it indefinite, right?